packages/design-system/src/components/ChoiceList/Choice.jsx

```diff
@@ -13,9 +13,10 @@ export class Choice extends React.PureComponent {
     super(props);
     this.input = null;
     this.handleChange = this.handleChange.bind(this);
+    this.handleUncheck = this.handleUncheck.bind(this);
     this.id = this.props.id || uniqueId(`${this.props.type}_${this.props.name}_`);
 
-    if (typeof this.props.checked === 'undefined') {
+    if (typeof this.props.checked === 'undefined' || this.props.defaultChecked) {
```

Comment:
Sorry, this code wont actually work. This will set the component to controlled even if a user passes in `defaultChecked = true`. You should just remove this change I believe